IOSDeviceLib/Printing.cpp

```diff
@@ -1,16 +1,24 @@
+#include <mutex>
+
 #include "SocketHelper.h"
 #include "Printing.h"
 #include "Constants.h"
 #include "json.hpp"
 
+std::mutex print_mutex;
+
 void print(const char* str)
 {
+	// We need to lock the print method because in some cases we print different parts of messages
+	// from different threads.
+	print_mutex.lock();
```

Comment:
I prefer using the mutex whenever we print on the terminal as this way we'll guarantee correct order of messages even if only part of them is sent to stdout's on data event